File: lt-api/gateway/group.go
package gateway

import (
	"main/domain"
	"main/driver"
)

type GroupGateway struct {
	groupDriver driver.GroupDriver
}

func (t GroupGateway)GetAll() ([]domain.Group,error){
	result, err := t.groupDriver.GetAll()

	if err != nil {
		return nil, err
	}

	var groups []domain.Group

	for _, t := range result {
		group := domain.Group{
			Id: domain.GroupId{Value: t.Id},
			Name: domain.GroupName{Value: t.Name},
			Description: domain.GroupDescription{Value: t.Description},
			Date: domain.GroupDate{Value: t.Date},
		}
		groups = append(groups, group)
	}

	return groups,nil
}

func ProvideGroupPort(groupDriver driver.GroupDriver) GroupGateway {
	return GroupGateway{groupDriver}
}

File: lt-front/src/lib/domain/Group.ts
export class Group {
  constructor(
    readonly id: GroupId,
    readonly name: GroupName,
    readonly description: GroupDescription,
    readonly date: GroupDate
  ) {}
}

export class GroupId {
  constructor(readonly value: number) {}
}

export class GroupName {
  constructor(readonly value: string) {}
}

export class GroupDescription {
  constructor(readonly value: string) {}
}

export class GroupDate {
  constructor(readonly value: Date) {}
}
